test/jdk/javax/swing/text/html/StyleSheet/TestExternalCSSFontSize.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import java.awt.Component;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import javax.imageio.ImageIO;
+import javax.swing.JEditorPane;
+import javax.swing.SwingUtilities;
+import javax.swing.text.GlyphView;
+import javax.swing.text.View;
+
+/*
+ * @test
+ * @bug 8292948
+ * @summary  Tests font-size declarations from external style sheet.
+ * @run main TestExternalCSSFontSize
+ */
+public class TestExternalCSSFontSize {
+
+    private static final int[] expectedFontSizes = { 21, 14, 12 };
+
+    private JEditorPane editor;
+
+    private volatile Throwable failure;
+
+    TestExternalCSSFontSize() {}
+
+    CountDownLatch setUp() throws Exception {
+        File htmlFile = new File("TestExternalCSSFontSize.html");
```

Comment:
Not sure if I should better resolve this from the classpath:

```suggestion
        URL htmlFile = getClass().getResource(getClass().getSimpleName() + ".html");
```